scripts/docker_entrypoint.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -e # stop script execution if any errors are encountered
+
+# the path to the script directory. Used for relative paths
+SCRIPTS_PATH=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd)
+
+# script help info
+function usage() {
+    cat <<USAGE
+
+    Usage: $0 [--rebuild_kb <PATH>] [sc-server args]
+
+    Options:
+        build <PATH>:       rebuilds KB from sources (provide absolute path to the source folder or repo.path file)
+        serve <args>:       Starts sc-server. Arguments passed to this command will be redirected to sc-server binary. If no arguments were given, uses "-h 0.0.0.0 -c /sc-machine/config/sc-machine.ini" as default sc-server arguments. Add these settings yourself if you are planning to use custom arguments.
+
+        Setting REBUILD_KB evironment variable inside the container will trigger a KB rebuild. Setting custom starting point for build_kb.py can be done using KB_PATH environment variable, "/kb" is used as a default KB_PATH.
+
+USAGE
+    "$SCRIPTS_PATH"/../bin/sc-server --help
+    exit 1
+}
+
+function rebuild_kb() {
+    if [ -e "$1" ]; then
+        python3 "$SCRIPTS_PATH/build_kb.py" -c "$SCRIPTS_PATH/../config/sc-machine.ini" "$1"
+    elif [ -z "$KB_PATH" ]; then
+        echo "$KB_PATH is set as a KB path by the environment variable"
+        python3 "$SCRIPTS_PATH/build_kb.py" -c "$SCRIPTS_PATH/../config/sc-machine.ini" "$KB_PATH" 
+    else
+        echo "Invalid KB source path provided."
+        exit 1
+    fi
+}
+
+function start_server() {
+if [ "$REBUILD_KB" -eq 1 ]; then
```

Comment:
Which values can this var take?

Reply:
Theoretically it can contain any value, but it doesn't matter unless it is set to 1. We can also check if the variable is set at all, but I decided to make it a bit more explicit in case we'll flip the default and we will build the KB on relaunch. You'll just have to set it to any value except 1 (e.g 0) to disable this.

Reply:
maybe it's better to use some kind of identifier rather than a number?